src/Mvc/Mvc.Core/src/JsonOptions.cs

```diff
@@ -26,6 +27,9 @@ public class JsonOptions
 
             // Use camel casing for properties
             PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+
+            // Does not encode all non-ASCII characters which matches Json.NET"s behavior.
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
```

Comment:
I believe this is a dangerous place to put this line of code. The _UnsafeRelaxedJsonEscaping_ property is __only__ intended to be used in specific scenarios where the stack has tight control over how the data will be written to the wire and how the client will consume the data. Any other usage is dangerous, hence the term "unsafe" in the name.

Here, the part that causes me concern is that ASP.NET is configuring this setting in a place very far removed from its intended known-good usage scenario. Specifically, this particular escaping should only be set if we know for absolute certain that the response will contain a header `Content-Type: application/json; charset=utf=8`, but this line of code is so far removed from the component that sets the response Content-Type header that this risks introducing a vulnerability due to misuse.

Reply:
What the heck is the point in having safe defaults if asp.net is going to dump all over them and turn it off?

Reply:
We might be able to localize this to the Json Formatter when the encoding is Uf8 and leave the defaults untouched. Would that be sufficient?